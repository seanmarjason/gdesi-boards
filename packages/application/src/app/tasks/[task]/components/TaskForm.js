import { useState } from "react";
import dayjs from 'dayjs';

import Typography from '@mui/material/Typography';
import TextField from '@mui/material/TextField';
import InputLabel from '@mui/material/InputLabel';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import Chip from '@mui/material/Chip';
import Avatar from '@mui/material/Avatar';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import Button from '@mui/material/Button';
import Link from '@mui/material/Link';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';

import { DatePicker } from '@mui/x-date-pickers/DatePicker'
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';

import { taskTypeMenuItems } from '../../../../data/tasks';
import { boardUsers } from '../../../../data/boardUsers';

export const TaskForm = ({ task }) => {

    const emptyLink = {name: '', type: '', url: ''}
    const emptyComment = {author: '', dateCreated: '', dateModified: '', comment: ''}

    const [deadline, setDeadline] = useState(dayjs(task.deadline));

    const [title, setTitle]  = useState(task.title)
    const [type, setType]  = useState(task.type)
    const [assignee, setAssignee]  = useState(task.assignee)
    const [description, setDescription]  = useState(task.description)
    const [links, setLinks]  = useState(task.links)
    const [comments, setComments]  = useState(task.comments)

    const [showLinkModal, setShowLinkModal] = useState(false)
    const [linkModalDetails, setLinkModalDetails] = useState(emptyLink)
    const [editLink, setEditLink] = useState(false)

    const [newComment, setNewComment] = useState(emptyComment)

    const statusColours = {
        'To Do': '',
        'Doing': 'primary',
        'Done': 'success',
    }

    const saveLinkDetails = (newLink) => {
        let nextLinks;

        // check if newLink id in links
        if (links.some(link => link.id == newLink.id)) {
            // if yes, update links
            nextLinks = links.map(link => link.id == newLink.id ? newLink : link)
        }
        else {
            // if no, add newLink to links
            nextLinks = [
                ...links,
                newLink
            ]
        }

        setLinks(nextLinks)
        setEditLink(false)
        setLinkModalDetails(emptyLink)
    }

    const saveComment = (comment) => {
        const user = 'Mr Miyagi'
        const date = new Date(Date.now()).toISOString()

        setComments([
            ...comments,
            {
                ...comment,
                author: user,
                dateCreated: date,
                dateModified: date
            }
        ])

        setNewComment(emptyComment)
    }

    return (
        <div>
            <Typography variant="h1" gutterBottom>
                {task.id}
            </Typography>

            <form>

                <Chip label={task.status} color={statusColours[task.status]} variant="outlined" />
                
                <br />

                <InputLabel id="task-deadline-label">Deadline</InputLabel>
                <LocalizationProvider dateAdapter={AdapterDayjs}>
                    <DatePicker 
                        value={deadline}
                        disablePast
                        onAccept={(value) => setDeadline(value)}
                    />
                </LocalizationProvider>

                <InputLabel id="task-title-label">Title</InputLabel>
                <TextField
                    id="task-title"
                    value={title}
                    fullWidth
                    onChange={(event) => {
                        setTitle(event.target.value)
                    }}
                />

                <InputLabel id="select-type-label">Type</InputLabel>
                <Select
                    labelId="select-type-label"
                    id="select-type"
                    value={type}
                    label="Type"
                    onChange={(event) => {
                        setType(event.target.value)
                    }}
                >
                    {taskTypeMenuItems.map((item, index) => 
                        <MenuItem value={item} key={`select-type-${index}`}>{item}</MenuItem>
                    )}
                </Select>

                <InputLabel id="select-assignee-label">Assignee</InputLabel>
                <Select
                    labelId="select-assignee-label"
                    id="select-assignee"
                    value={assignee}
                    label="Assignee"
                    onChange={(event) => {
                        setAssignee(event.target.value)
                    }}
                >
                    {boardUsers.map((item, index) => 
                        <MenuItem value={item} key={`select-assignee-${index}`}>{item}</MenuItem>
                    )}
                </Select>

                <InputLabel id="task-description-label">Description</InputLabel>
                <TextField
                    fullWidth
                    multiline
                    // minRows={4} //TODO: Fix styling causing multiline to not expand
                    id="task-description"
                    value={description}
                    onChange={(event) => {
                        setDescription(event.target.value)
                    }}
                />
                
                <InputLabel id="task-links-label">Links</InputLabel>
                { links.length > 0 ?
                    (links.map((link, index) => 
                        <Chip
                            key={`link-${index}`}
                            label={link.name}
                            avatar={<Avatar>{(Array.from(link.name)[0])?.toUpperCase()}</Avatar>}
                            onClick={() => {
                                setShowLinkModal(true)
                                setLinkModalDetails(link)
                            }}
                        />
                    ))
                    : <p>nil</p>
                }

                <Chip
                    key={`link-new`}
                    label={'New'}
                    avatar={<Avatar>+</Avatar>}
                    onClick={() => {
                        setShowLinkModal(true)
                        setEditLink(true)
                        setLinkModalDetails(emptyLink)
                    }}
                />

                {   showLinkModal &&

                        <Dialog
                            open={showLinkModal}
                            onClose={() => setShowLinkModal(false)}
                            aria-labelledby="modal-modal-title"
                            aria-describedby="modal-modal-description"
                        >

                        {
                            editLink ?
                             <div>                            
                                <DialogTitle id="alert-dialog-title">New</DialogTitle>
                                    <InputLabel id="link-title-label">Name</InputLabel>
                                    <TextField
                                        id="link-name"
                                        value={linkModalDetails.name}
                                        onChange={(event) => {
                                            setLinkModalDetails({
                                                ...linkModalDetails,
                                                name: event.target.value,
                                            })
                                        }}
                                    />
                                    <InputLabel id="link-type-label">Type</InputLabel>
                                    <TextField
                                        id="link-type"
                                        value={linkModalDetails.type}
                                        onChange={(event) => {
                                            setLinkModalDetails({
                                                ...linkModalDetails,
                                                type: event.target.value,
                                            })
                                        }}
                                    />
                                    <InputLabel id="link-type-label">Url</InputLabel>
                                    <TextField
                                        id="link-url"
                                        value={linkModalDetails.url}
                                        onChange={(event) => {
                                            setLinkModalDetails({
                                                ...linkModalDetails,
                                                url: event.target.value,
                                            })
                                        }}
                                    />
                                    <DialogActions>
                                        <Button 
                                            onClick={() => {
                                                setShowLinkModal(false) 
                                                saveLinkDetails({
                                                    ...linkModalDetails,
                                                    id: linkModalDetails.id || `${task.id}_link-${links.length + 1}`
                                                }
                                            )}
                                            }
                                            autoFocus
                                        >
                                        Save
                                        </Button>
                                    </DialogActions>
                            </div>
                            :
                            <div>                            
                                <DialogTitle id="alert-dialog-title">{linkModalDetails.name}</DialogTitle>
                                <DialogContent>
                                    <DialogContentText id="alert-dialog-description">
                                        {linkModalDetails.type}
                                        <br />
                                        <Link href={linkModalDetails.url} target="_blank">
                                            {linkModalDetails.url}
                                        </Link>
                                    </DialogContentText>
                                </DialogContent>
                                <DialogActions>
                                    <Button onClick={() => setEditLink(true)}>Edit</Button>
                                    <Button onClick={() => setShowLinkModal(false)} autoFocus>Ok</Button>
                                </DialogActions>
                            </div>
                        }
                        </Dialog>
                }
                
                <InputLabel id="task-comments-label">Comments</InputLabel>
                { comments.length > 0 ?
                    (comments.map((comment, index) => 
                        <Card
                            sx={{ minWidth: 275 }}
                            key={`comment-${index}`}
                        >
                            <CardContent>
                                <Typography sx={{ fontSize: 14 }}>
                                    {comment.author}
                                </Typography>
                                <Typography sx={{ color: 'text.secondary', mb: 1.5 }}>
                                    Created: {comment.dateCreated} Modified: {comment.dateModified}
                                </Typography>
                                <Typography variant="body2">
                                    {comment.comment}
                                </Typography>
                            </CardContent>
                        </Card>
                    ))
                    : <p>nil</p>
                }

                <InputLabel id="new-comment-label">Add New Comment</InputLabel>
                    <TextField
                        id="new-comment"
                        value={newComment.comment}
                        onChange={(event) => {
                            setNewComment({
                                ...newComment,
                                comment: event.target.value,
                            })
                        }}
                    />
                    <Button onClick={() => saveComment(newComment)}>Save</Button>

            </form>

        </div>
    )
}