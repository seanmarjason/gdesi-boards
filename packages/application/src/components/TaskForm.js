import { useState } from "react";
import dayjs from 'dayjs';
import 'dayjs/locale/en-gb';

import NextLink from 'next/link';

import Typography from '@mui/material/Typography';
import TextField from '@mui/material/TextField';
import InputLabel from '@mui/material/InputLabel';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import Chip from '@mui/material/Chip';
import Avatar from '@mui/material/Avatar';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import Button from '@mui/material/Button';
import Link from '@mui/material/Link';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';

import { DatePicker } from '@mui/x-date-pickers/DatePicker'
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';

import { taskTypeMenuItems } from '../data/tasks';

export const TaskForm = ({ task, boardId, users, currentUser, saveTask, checkComment }) => {

    const emptyLink = {name: '', type: '', url: ''}
    const emptyComment = {author: '', dateCreated: '', comment: ''}

    const [deadline, setDeadline] = useState(task?.deadline ? dayjs(task?.deadline) : dayjs().add(7, 'day'));

    const [title, setTitle]  = useState(task?.title ?? '')
    const [type, setType]  = useState(task?.type ?? '')
    const [assignee, setAssignee]  = useState(task?.assignee ?? '')
    const [description, setDescription]  = useState(task?.description ?? '')
    const [links, setLinks]  = useState(task?.links ?? [])
    const [comments, setComments]  = useState(task?.comments ?? [])

    const [showLinkModal, setShowLinkModal] = useState(false)
    const [linkModalDetails, setLinkModalDetails] = useState(emptyLink)
    const [editLink, setEditLink] = useState(false)

    const [newComment, setNewComment] = useState(emptyComment)
    const [showCommentLoading, setShowCommentLoading] = useState(false)

    const statusColours = {
        'To Do': '',
        'Doing': 'primary',
        'Done': 'success',
    }

    const saveLinkDetails = (newLink) => {
        let nextLinks;

        // check if newLink id in links
        if (links.some(link => link.id == newLink.id)) {
            // if yes, update links
            nextLinks = links.map(link => link.id == newLink.id ? newLink : link)
        }
        else {
            // if no, add newLink to links
            nextLinks = [
                ...links,
                newLink
            ]
        }

        setLinks(nextLinks)
        setEditLink(false)
        setLinkModalDetails(emptyLink)
    }

    const saveComment = async (comment) => {
        const date = new Date(Date.now()).toISOString()

        setComments([
            ...comments,
            {
                ...comment,
                author: currentUser,
                dateCreated: date,
            },
        ])

        setNewComment(emptyComment)

        setShowCommentLoading(true)
        const commentEvaluation = await checkComment(comment.comment)

        setComments([
            ...comments,
            {
                ...comment,
                author: currentUser,
                dateCreated: date,
            },
            {
                author: 'BOT', 
                dateCreated: date, 
                comment: commentEvaluation
            }
        ])
        setShowCommentLoading(false)
    }

    return (
        <div>
            <Typography variant="h1" gutterBottom>
                {task?.id ?? 'New Task'}
            </Typography>

            <form>

                <Chip label={task?.status ?? 'To Do'} color={statusColours[task?.status]} variant="outlined" />
                
                <br />

                <InputLabel id="task-deadline-label">Deadline</InputLabel>
                <LocalizationProvider dateAdapter={AdapterDayjs} adapterLocale={'en-gb'}>
                    <DatePicker 
                        value={deadline}
                        disablePast
                        onAccept={(value) => setDeadline(value)}
                    />
                </LocalizationProvider>

                <InputLabel id="task-title-label">Title</InputLabel>
                <TextField
                    id="task-title"
                    value={title}
                    fullWidth
                    onChange={(event) => {
                        setTitle(event.target.value)
                    }}
                />

                <InputLabel id="select-type-label">Type</InputLabel>
                <Select
                    labelId="select-type-label"
                    id="select-type"
                    value={type}
                    label="Type"
                    onChange={(event) => {
                        setType(event.target.value)
                    }}
                >
                    {taskTypeMenuItems.map((item, index) => 
                        <MenuItem value={item} key={`select-type-${index}`}>{item}</MenuItem>
                    )}
                </Select>

                <InputLabel id="select-assignee-label">Assignee</InputLabel>
                <Select
                    labelId="select-assignee-label"
                    id="select-assignee"
                    value={assignee}
                    label="Assignee"
                    onChange={(event) => {
                        setAssignee(event.target.value)
                    }}
                >
                    {users.map((user, index) => 
                        <MenuItem value={user.id} key={`select-assignee-${index}`}>{user.name}</MenuItem>
                    )}
                </Select>

                <InputLabel id="task-description-label">Description</InputLabel>
                <TextField
                    fullWidth
                    multiline
                    // minRows={4} //TODO: Fix styling causing multiline to not expand
                    id="task-description"
                    value={description}
                    onChange={(event) => {
                        setDescription(event.target.value)
                    }}
                />
                
                <InputLabel id="task-links-label">Links</InputLabel>
                { links.length > 0 ?
                    (links.map((link, index) => 
                        <Chip
                            key={`link-${index}`}
                            label={link.name}
                            avatar={<Avatar>{(Array.from(link.name)[0])?.toUpperCase()}</Avatar>}
                            onClick={() => {
                                setShowLinkModal(true)
                                setLinkModalDetails(link)
                            }}
                        />
                    ))
                    : <p>nil</p>
                }

                <Chip
                    key={`link-new`}
                    label={'New'}
                    avatar={<Avatar>+</Avatar>}
                    onClick={() => {
                        setShowLinkModal(true)
                        setEditLink(true)
                        setLinkModalDetails(emptyLink)
                    }}
                />

                {   showLinkModal &&

                        <Dialog
                            open={showLinkModal}
                            onClose={() => setShowLinkModal(false)}
                            aria-labelledby="modal-modal-title"
                            aria-describedby="modal-modal-description"
                        >

                        {
                            editLink ?
                             <div>                            
                                <DialogTitle id="alert-dialog-title">New</DialogTitle>
                                    <InputLabel id="link-title-label">Name</InputLabel>
                                    <TextField
                                        id="link-name"
                                        value={linkModalDetails.name}
                                        onChange={(event) => {
                                            setLinkModalDetails({
                                                ...linkModalDetails,
                                                name: event.target.value,
                                            })
                                        }}
                                    />
                                    <InputLabel id="link-type-label">Type</InputLabel>
                                    <TextField
                                        id="link-type"
                                        value={linkModalDetails.type}
                                        onChange={(event) => {
                                            setLinkModalDetails({
                                                ...linkModalDetails,
                                                type: event.target.value,
                                            })
                                        }}
                                    />
                                    <InputLabel id="link-type-label">Url</InputLabel>
                                    <TextField
                                        id="link-url"
                                        value={linkModalDetails.url}
                                        onChange={(event) => {
                                            setLinkModalDetails({
                                                ...linkModalDetails,
                                                url: event.target.value,
                                            })
                                        }}
                                    />
                                    <DialogActions>
                                        <Button 
                                            onClick={() => {
                                                setShowLinkModal(false) 
                                                saveLinkDetails({
                                                    ...linkModalDetails,
                                                    id: linkModalDetails.id || `${task.id}_link-${links.length + 1}`
                                                }
                                            )}
                                            }
                                            autoFocus
                                        >
                                        Save
                                        </Button>
                                    </DialogActions>
                            </div>
                            :
                            <div>                            
                                <DialogTitle id="alert-dialog-title">{linkModalDetails.name}</DialogTitle>
                                <DialogContent>
                                    <DialogContentText id="alert-dialog-description">
                                        {linkModalDetails.type}
                                        <br />
                                        <Link href={linkModalDetails.url} target="_blank">
                                            {linkModalDetails.url}
                                        </Link>
                                    </DialogContentText>
                                </DialogContent>
                                <DialogActions>
                                    <Button onClick={() => setEditLink(true)}>Edit</Button>
                                    <Button onClick={() => setShowLinkModal(false)} autoFocus>Ok</Button>
                                </DialogActions>
                            </div>
                        }
                        </Dialog>
                }
                
                <InputLabel id="task-comments-label">Comments</InputLabel>
                { comments.length > 0 ?
                    (comments.map((comment, index) => 
                        <Card
                            sx={{ minWidth: 275 }}
                            key={`comment-${index}`}
                        >
                            <CardContent>
                                <Typography sx={{ fontSize: 14 }}>
                                    {comment.author}
                                </Typography>
                                <Typography sx={{ color: 'text.secondary', mb: 1.5 }}>
                                    Created: {comment.dateCreated}
                                </Typography>
                                <Typography variant="body2">
                                    {comment.comment}
                                </Typography>
                            </CardContent>
                        </Card>
                    ))
                    : <p>nil</p>
                }

                { showCommentLoading && 
                    <Card
                        sx={{ minWidth: 275 }}
                        key={`comment-loading`}
                    >
                        <CardContent>
                            <Typography variant="body2">
                                Loading ...
                            </Typography>
                        </CardContent>
                    </Card>
                }

                <InputLabel id="new-comment-label">Add New Comment</InputLabel>
                    <TextField
                        id="new-comment"
                        value={newComment.comment}
                        onChange={(event) => {
                            setNewComment({
                                ...newComment,
                                comment: event.target.value,
                            })
                        }}
                    />
                    <Button onClick={() => saveComment(newComment)}>Add</Button>

            <br />
            <br />


            <Button component={NextLink} href={`/boards/${boardId}`} variant="contained" color="error">Cancel</Button>
            <Button 
                variant="contained" 
                color="secondary"
                onClick={() => saveTask({
                    deadline,
                    title,
                    type,
                    assignee,
                    description,
                    links,
                    comments,
                })} 
            >
                Save
            </Button>

            </form>

        </div>
    )
}