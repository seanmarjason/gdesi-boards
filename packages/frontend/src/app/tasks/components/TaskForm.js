
export const TaskForm = ({}) => {
    return (
        <p>TaskForm</p>
    )
}