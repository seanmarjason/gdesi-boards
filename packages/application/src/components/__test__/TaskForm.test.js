import { render } from '@testing-library/react'
import { TaskForm } from '../TaskForm';

jest.mock("@mui/x-date-pickers/DatePicker");
jest.mock("@mui/x-date-pickers/LocalizationProvider");
jest.mock("@mui/x-date-pickers/AdapterDayjs");
jest.mock("dayjs");

const task = {
    status: 'status',
    title: 'title',
    deadline: 'deadline',
    type: 'Document',
    assignee: 'assignee',
    description: 'description',
    links: [],
    comments: [],
}
 
it('TaskForm renders correctly', () => {
  const { container } = render(
    <TaskForm 
        task={task}
        boardId={1}
        users={[{id: 1, name: 'user'}]}
        currentUser={'user'}
        saveTask={jest.fn()}
        checkComment={jest.fn()}
    />
  )
  expect(container).toMatchSnapshot()
})
