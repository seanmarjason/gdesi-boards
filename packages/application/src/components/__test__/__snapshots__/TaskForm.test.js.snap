// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-dyz9rb-MuiTypography-root"
    >
      New Task
    </h1>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <form
        sx="[object Object]"
      >
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1lrxra9-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
          >
            status
          </span>
        </div>
        <br />
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
          id="task-deadline-label"
        >
          Deadline
        </label>
        <p
          class="MuiFormHelperText-root Mui-error css-18y2hez-MuiFormHelperText-root"
        >
          Required
        </p>
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
          id="task-title-label"
        >
          Title
        </label>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
              id="task-title"
              type="text"
              value="title"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-w4cd9x"
              >
                <span
                  aria-hidden="true"
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
          id="select-type-label"
        >
          Type
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiSelect-root css-sc8y68-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="select-type-label select-type"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="select-type"
            role="combobox"
            tabindex="0"
          >
            Document
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
            tabindex="-1"
            value="Document"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-81qg8w"
            >
              <span>
                Type
              </span>
            </legend>
          </fieldset>
        </div>
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
          id="select-assignee-label"
        >
          Assignee
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiSelect-root css-sc8y68-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="select-assignee-label select-assignee"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="select-assignee"
            role="combobox"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
            tabindex="-1"
            value="assignee"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-81qg8w"
            >
              <span>
                Assignee
              </span>
            </legend>
          </fieldset>
        </div>
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
          id="task-description-label"
        >
          Description
        </label>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-zsp0bh-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
              id="task-description"
              style="height: 0px; overflow: hidden;"
            >
              description
            </textarea>
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-w4cd9x"
              >
                <span
                  aria-hidden="true"
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
          id="task-links-label"
        >
          Links
        </label>
        <p>
          nil
        </p>
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-18k2bs-MuiAvatar-root"
          >
            +
          </div>
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            New
          </span>
        </div>
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
          id="task-comments-label"
        >
          Comments
        </label>
        <p>
          nil
        </p>
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
          id="new-comment-label"
        >
          Add New Comment
        </label>
        <div
          class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
              id="new-comment"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-w4cd9x"
              >
                <span
                  aria-hidden="true"
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
          id=":r5:"
          tabindex="0"
          type="button"
        >
          Add
        </button>
        <br />
        <br />
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError css-1f7qp0x-MuiButtonBase-root-MuiButton-root"
          href="/boards/1"
          id=":r6:"
          tabindex="0"
        >
          Cancel
        </a>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1mns5ui-MuiButtonBase-root-MuiButton-root"
          id=":r7:"
          tabindex="0"
          type="button"
        >
          Save
        </button>
      </form>
    </div>
  </div>
</div>
`;
